compiler/src/dotty/tools/dotc/interactive/Completion.scala

```diff
@@ -131,15 +142,27 @@ object Completion {
     /**
      * Return the list of symbols that shoudl be included in completion results.
      *
-     * If the mode is `Import` and several symbols share the same name, the type symbols are
-     * preferred over term symbols.
+     * If several symbols share the same name, the type symbols appear before term symbols inside
+     * the same `Completion`.
+     */
+    def getCompletions(implicit ctx: Context): List[Completion] = {
+      val groupedSymbols = completions.toList.groupBy(_.name.stripModuleClassSuffix.toSimpleName).toList
+      groupedSymbols.map { case (name, symbols) =>
+        val typesFirst = symbols.sortWith((s, _) => s.isType)
+        // Use distinct to remove duplicates with class, module class, etc.
+        val descriptions = typesFirst.map(description).distinct.mkString(", ")
```

Comment:
I would use `" and "` since `,` could be confused with type parameter lists and tuples